app/controllers/bike_share_app.rb

```diff
@@ -1,3 +1,131 @@
+require 'will_paginate'
+require 'will_paginate/active_record'
+
 class BikeShareApp < Sinatra::Base
+include WillPaginate::Sinatra::Helpers
+
+  get '/' do
+    erb :homepage
+  end
+
+  get '/stations' do
+    @stations = Station.all
+    erb :"stations/index"
+  end
+
+  get '/stations/new' do
+    erb :"/stations/new"
+  end
+
+  get '/stations-dashboard' do
+    erb :"/stations/dashboard"
+  end
+
+  get '/stations/:id' do
+    @station = Station.find(params[:id])
+    erb :"stations/show"
+  end
+
+  post '/stations' do
+    station = Station.write(params[:stations])
+    redirect "/stations/#{ station.id }"
+  end
+
+  get '/stations/:id/edit' do
+    @station = Station.find(params[:id])
+    erb :"stations/edit"
+  end
+
+  put '/stations/:id' do
+    station = Station.find(params[:id])
+    station.write_update(station, params[:stations])
+    redirect "stations/#{station.id}"
+  end
+
+  delete '/stations/:id' do
+    Station.delete(params[:id])
+    redirect "/stations"
+  end
+
+  get '/trips' do
+    @trips = Trip.paginate(:page => params[:page], :per_page => 30)
+    erb :'trips/index'
+  end
+
+  get '/trips/new' do
+    erb :'/trips/new'
+  end
+
+  post '/trips' do
+    trip = Trip.write(params[:trips])
+    redirect "/trips/#{ trip.id }"
+  end
+
+  get '/trips-dashboard' do
+    erb :'/trips/dashboard'
+  end
+
+  get '/trips/:id/edit' do
+    @trip = Trip.find(params[:id])
+    @subscription_type = Subscription.find(@trip.subscription_id)
```

Comment:
if trip has an association with subscription, we should be able to just call `@trip.subscription` rather than calling `find`